crates/subspace-farmer/src/bin/subspace-farmer/commands/cluster/farmer.rs

```diff
@@ -0,0 +1,515 @@
+    ///
+    ///   path=/path/to/directory,size=5T
+    ///
+    /// `size` is max allocated size in human-readable format (e.g. 10GB, 2TiB) or just bytes that
+    /// farmer will make sure to not exceed (and will pre-allocated all the space on startup to
+    /// ensure it will not run out of space in runtime). Optionally, `record-chunks-mode` can be
+    /// set to `ConcurrentChunks` or `WholeSector` in order to avoid internal benchmarking during
+    /// startup.
+    disk_farms: Vec<DiskFarm>,
+    /// Address for farming rewards
+    #[arg(long, value_parser = parse_ss58_reward_address)]
+    reward_address: PublicKey,
+    /// Run temporary farmer with specified farm size in human-readable format (e.g. 10GB, 2TiB) or
+    /// just bytes (e.g. 4096), this will create a temporary directory that will be deleted at the
+    /// end of the process.
+    #[arg(long, conflicts_with = "disk_farms")]
+    tmp: Option<ByteSize>,
+    /// Maximum number of pieces in sector (can override protocol value to something lower).
+    ///
+    /// This will make plotting of individual sectors faster, decrease load on CPU proving, but also
+    /// proportionally increase amount of disk reads during audits since every sector needs to be
+    /// audited and there will be more of them.
+    ///
+    /// This is primarily for development and not recommended to use by regular users.
+    #[arg(long)]
+    max_pieces_in_sector: Option<u16>,
+    /// Do not print info about configured farms on startup
+    #[arg(long)]
+    no_info: bool,
+    /// Defines max number sectors farmer will encode concurrently, defaults to 8. Might be limited
+    /// by plotting capacity available in the cluster.
+    ///
+    /// Increase will result in higher memory usage.
+    #[arg(long, default_value = "8")]
+    sector_encoding_concurrency: NonZeroUsize,
+    /// Size of PER FARM thread pool used for farming (mostly for blocking I/O, but also for some
+    /// compute-intensive operations during proving), defaults to number of logical CPUs
+    /// available on UMA system and number of logical CPUs in first NUMA node on NUMA system, but
+    /// not more than 32 threads
+    #[arg(long)]
+    farming_thread_pool_size: Option<NonZeroUsize>,
+    /// Disable farm locking, for example if file system doesn't support it
+    #[arg(long)]
+    disable_farm_locking: bool,
+    /// Whether to create missing farms during start.
+    ///
+    /// If set to `false` farmer will exit with error if one of the farms doesn't already exist.
+    #[arg(long, default_value_t = true, action = clap::ArgAction::Set)]
+    create: bool,
+    /// Exit on farm error.
+    ///
+    /// By default, farmer will continue running if there are still other working farms.
+    #[arg(long)]
+    exit_on_farm_error: bool,
+    /// Additional cluster components
+    #[clap(raw = true)]
+    pub(super) additional_components: Vec<String>,
+}
+
+pub(super) async fn farmer<PosTable>(
```

Comment:
I have to agree, but so is `farm` method that this is largely based on. I'd rather look at them both and refactor them together. In fact the code here is almost identical to `farm` with the difference being that node client and plotter are remote rather than local and plotted/cached contents no longer needs to be maintained because controller does it in this case.